modules/smtp/modules.php

```diff
@@ -888,6 +904,97 @@ protected function output() {
     }
 }
 
+/**
+ * Send scheduled messages
+ * @subpackage smtp/handler
+ */
+class Hm_Handler_send_scheduled_messages extends Hm_Handler_Module {
+    /**
+     * Send delayed messages
+     * This should use cron
+     */
+   public function process() {
+    if (!($this->module_is_supported('imap') || $this->module_is_supported('profiles'))) {
+        return;
+    }
+
+    $servers = Hm_IMAP_List::dump();
+    $scheduled_msg_count = 0;
+
+    foreach (array_keys($servers) as $server_id) {
+        $cache = Hm_IMAP_List::get_cache($this->cache, $server_id);
+        $imap = Hm_IMAP_List::connect($server_id, $cache);
+
+        if (imap_authed($imap)) {
+            $folder = 'Scheduled';
+            $ret = $imap->get_mailbox_page($folder, 'DATE', false, 'ALL');
+
+            foreach ($ret[1] as $msg) {
+                $msg_headers = $imap->get_message_headers($msg['uid']);
+
+                try {
+                    if (!empty($msg_headers['X-Schedule'])) {
+                        $scheduled_msg_count++;
+                    } else {
+                        continue;
+                    }
+
+                    if (send_scheduled_message($this, $imap, $msg, $server_id)) {
+                        $scheduled_msg_count--;
+                    }
+                } catch (Exception $e) {
+                    Hm_Debug::add(sprintf('ERRCannot send message: %s', $msg_headers['subject']));
+                }
+            }
+        }
+    }
+
+    $this->out('scheduled_msg_count', $scheduled_msg_count);
+}}
+
+/**
+ * Changes the schedule of the message
+ * @subpackage smtp/handler
+ */
+class Hm_Handler_re_schedule_message_sending extends Hm_Handler_Module {
+    public function process() {        
+        if (!($this->module_is_supported('imap') || $this->module_is_supported('profiles'))) {
+            return;
+        }
+        list($success, $form) = $this->process_form(array('schedule_date', 'scheduled_msg_ids'));
+        if (!$success) {
+            return;
+        }
+        $scheduled_msg_count = 0;
+        $new_schedule_date = $form['schedule_date'];
+        if ($form['schedule_date'] != 'now') {
+            $new_schedule_date = get_scheduled_date($form['schedule_date']);
+        }       
+        $ids = explode(',', $form['scheduled_msg_ids']);
+        foreach ($ids as $msg_part) {
+            list($imap_server_id, $msg_id, $folder) = explode('_', $msg_part);
+            $cache = Hm_IMAP_List::get_cache($this->cache, $imap_server_id);
+            $imap = Hm_IMAP_List::connect($imap_server_id, $cache);
```

Comment:
Now that EWS has been merged to master, we bypass direct imap object usage from Hm_IMAP_List in all handler modules and instead use Hm_Mailbox which is a bridge between handlers and imap/ews implementation classes. Please update this code according to what is currently present in imap handler_modules after merging (or rebasing) latest mater.